app/util/nns.js

```diff
@@ -0,0 +1,9 @@
+import { rpc } from 'neon-js'
+
+export const resolveNnsDomain = name =>
+  rpc
+    .queryRPC('https://apiwallet.nel.group/api/mainnet', {
+      method: 'getresolvedaddress',
+      params: [name]
+    })
+    .then(data => data.result[0].data)
```

Comment:
use lodash get here in case the data has a different shape or else add a catch clause to capture any errors

Reply:
While looking into adding this, it turns out that logic becomes has the same effect whether or not I parse using lodash get.

```
const address = get(data, 'result[0].data')
if (!address) throw 'malformatted response'
return address
```

In the case that the data is not in the shape as expected, an error will occur, and the promise will look for a catch statement. In this case, we want this to happen so that the user of this method can provide their own catch statement in the case of an error. As seen here:
https://github.com/CityOfZion/neon-wallet/blob/eb4dbb4ec5eaed56fb509439965ed9b68901eb83/app/components/Send/SendPanel/SendRecipientList/SendRecipientListItem/index.jsx#L73

An alternative would be to return a `null` value in the case where there was an error parsing the response, but then this requires method users to have 2 separate places where they need to catch errors, in the `then` clause where the response argument equals `null`, and in a `catch` statement for when a name is not found. In either case, from the UI perspective, the result is that the user will be provided with the error that the name was not found.

For now, will leave this as-is, unless there may be a case I am missing.

Reply:
Makes sense to me